components/script/dom/node.rs

```diff
@@ -1353,6 +1353,16 @@ pub enum CloneChildrenFlag {
 
 fn as_uintptr<T>(t: &T) -> uintptr_t { t as *const T as uintptr_t }
 
+// Checks if parent element only cares about text nodes
+fn accepts_only_text_nodes(node: &Node, parent: &Node) -> bool {
+    let isTextNode = node.children_count() == 0;
+    match parent.type_id() {
+        NodeTypeId::Element(ElementTypeId::HTMLElement(HTMLElementTypeId::HTMLHeadingElement)) |
```

Comment:
There should be more elements here, like the B element. But I don't know how to make the comparison for those type of elements.